drivers/interrupt_controller/intc_intel_vtd.c

```diff
@@ -51,6 +51,7 @@ static void vtd_send_cmd(const struct device *dev,
 
 }
 
+/* TODO: For MSI, we need contigious IRTE entry */
```

Comment:
which it always is aready. Useless addition.

Reply:
Current PCI bus driver code assume the vtd_ictl_allocate_entries() always serialized and current logic break if it is not

Reply:
It's always allocating the next contiguous n_vector available, I don't see what you mean here. Nothing is broken.

- at first data->irte_num_used = 0
- you request 1 entry, you get back index 0, data->irte_num_used = 1
- then you request 4 entries, you get back index 1 (meaning that you got also index 2, 3  and 4), data->irte_num_used = 5
- then you request 16 entries, you get back index 5  (meaning that you got also index up to 20), data->irte_num_used = 21
- etc...

Unlike the x86 core vector allocation, there is no holes there due to priority.

Where you see a contiguity issue?